src/firmware_new/boards/frankie_v1/udp_multicast.c

```diff
@@ -0,0 +1,123 @@
+#include "udp_multicast.h"
+
+#include "firmware_new/proto/control_fw.pb.h"
+#include "lwip/api.h"
+#include "lwip/inet.h"
+#include "lwip/ip_addr.h"
+#include "lwip/memp.h"
+#include "lwip/opt.h"
+#include "lwip/pbuf.h"
+#include "lwip/sys.h"
+#include "lwip/tcp.h"
+#include "lwip/udp.h"
+#include "main.h"
+#include "pb.h"
+#include "pb_decode.h"
+#include "pb_encode.h"
+
+// TODO the messages are global for now, will need to be moved
+// when this file is properly integrated. This file only serves
+// as a reference implementation of reliable multicast with IPv6
+robot_ack ack       = robot_ack_init_zero;
+control_msg control = control_msg_init_zero;
+
+// this buffer is used to hold serialized proto. Allocated
+// on the data segment to avoid mallocing on every packet received
+static uint8_t buffer[robot_ack_size];
```

Comment:
Wouldn't it be stack-allocated?

Reply:
The buffer itself will be part of the programs data segment

Reply:
Sorry, my question was why we have this globally, rather then in the function?

Reply:
honestly im not even sure why, moved it